### DISCUSSION TITLE
Is there a way to remove a Global DelegatingHandler from a specific Route?

### DISCUSSION THREAD

**Discussion by henrysmith** [ThreeMammals/Ocelot#1958] [Q&A] [answered]
I registered an auth global DelegatingHandler that applies to my routes.  However, there are a couple of routes that I need to avoid this handler.  My solution now is to set it as "no" global and add it route by route.

So, my question: is there a way to remove a Global DelegatingHandler from a specific Route?

If not, I think it would be a useful feature.

Henry

**Comment by raman-m** [comment]
@henrysmith Hi Henry!
>If not, I think it would be a useful feature.

Yes, it would!
**Will you contribute?
Will you create a PR?**

**Reply by raman-m** [reply, marked as answer]
Henry,
Because you developed custom delegating handler it will be better to exclude the routes right in the `SendAsync` method. If you've detected the route to exclude, just call `base.SendAsync` and the delegating handler will not process.
The collection of the routes to exclude you can save in **appsettings.json**, then inject `IOptions<TSection>` object into the constructor.
Why not to implement this solution?
